File: TwoWayBinding/Binding/BlockBasedSelector.m
//
//  BlockBasedSelector.m
//  TestMVVM
//
//  Created by Rohit Negi on 07/12/19.
//  Copyright © 2019 Rohit Negi. All rights reserved.
//

#import "BlockBasedSelector.h"
#import <objc/runtime.h>
@implementation BlockBasedSelector
@end
void class_addMethodWithBlock(Class class, SEL newSelector, OBJCBlock block){
    IMP newImplementation = imp_implementationWithBlock(block);
    Method method = class_getInstanceMethod(class, newSelector);
    class_addMethod(class, newSelector, newImplementation, method_getTypeEncoding(method));
}

File: TwoWayBinding/Binding/BlockBasedSelector.swift
//
//  BlockBasedSelector.swift
//  TestMVVM
//
//  Created by Rohit Negi on 07/12/19.
//  Copyright © 2019 Rohit Negi. All rights reserved.
//

import Foundation
import UIKit
@objc class _Selector: NSObject{
    static let shared = _Selector()
}
let Selector = _Selector.shared

func Selector(_ block: @escaping ()-> Void)-> Selector{
    let selector = NSSelectorFromString("\(CACurrentMediaTime())")
    class_addMethodWithBlock(_Selector.self, selector) { (_) in
        block()
    }
    return selector
}
